### ISSUE TITLE
miniconda silent install crashes Outlook

### ISSUE THREAD

**Issue by kalefranz** [ContinuumIO/anaconda-issues#968]
_From @davidanthoff on January 9, 2015 20:42_

When I do a silent miniconda install like this while I have Outlook 2013 x64 running

```
Miniconda3-3.7.0-Windows-x86.exe /AddToPath=0 /RegisterPython=0 /S /D=C:\somedir
```

it will crash Outlook. 100% reliably and reproducible. This is on Windows 8.1 x64.

This might be an old problem with NSIS that others have [had](http://sourceforge.net/p/nsis/bugs/1049/) [before](http://forums.winamp.com/showthread.php?t=260624).

_Copied from original issue: conda/conda#1084_

**Comment by kalefranz** [comment]
_From @stevengj on September 26, 2015 3:16_

See the comments [here](http://stackoverflow.com/questions/5317448/nsis-script-that-installs-chrome-extension-kills-outlook), which seem to suggest it was a problem with ZipDLL.

**Comment by kalefranz** [comment]
_From @stevengj on September 26, 2015 3:19_

It seems like you should comment on the NSIS bug if you can provide them with a simple reproducible case.

**Comment by kalefranz** [comment]
_From @davidanthoff on September 26, 2015 4:26_

@ilanschnell or @asmeurer, could you take a look at this issue? This is causing problems downstream with julia, where the conda package manager is now used by IJulia and a plotting package to install dependencies, but now crashes Outlook on Windows when the miniconda setup is run.

I would report this bug upstream with NSIS, but the conda repo does not seem to contain any setup logic, so I don't know what version of NSIS is used for the setup, nor do I have access to the setup script so that I could post that at the NSIS site. I think my bug report would be poor, whereas you might be able to give a proper one.

Bug I do believe that my bug report here should be good enough for you to repo the bug.

**Comment by kalefranz** [comment]
_From @tkelman on December 8, 2015 22:34_

Bump. Any triage here?

**Comment by kalefranz** [comment]
_From @ilanschnell on December 9, 2015 5:20_

Miniconda uses nsis 2.46 on Windows.

**Comment by kalefranz** [comment]
_From @stevengj on December 9, 2015 17:21_

(Note that the Conda package in Julia currently works around this by detecting whether Outlook is running and halting with an error message in that case, rather than crashing Outlook.  Luthaf/Conda.jl#15.)

@ilanschnell, if you can reproduce the Outlook crash, I agree with @davidanthoff that you are probably in a better position to add information to the NSIS bug because you know exactly how you set up NSIS.  We really appreciate your time and effort.

**Comment by kalefranz** [comment]
_From @brettcannon on July 26, 2016 20:54_

This is still a problem for Windows users. We just got a report today from someone using the miniconda 4.0.5 installer in silent mode crashing Outlook. And reporting this upstream to NSIS in hopes of a fix trickling down to miniconda is tough as [NSIS 2.46](https://sourceforge.net/projects/nsis/files/NSIS%202/2.46/) is over 6 years old and they released [NSIS 3.0 yesterday](http://nsis.sourceforge.net/News).

**Comment by stephenakearns** [comment]
Hi @csoja (cc: @kalefranz )
Unable to reproduce... 

Env: `Win10 Pro on Surface Pro 4, Outlook 2016. `
Miniconda: `Miniconda3-latest-Windows-x86_64.exe` (downloaded to day)
Command line: `c:\Miniconda3-latest-Windows-x86_64.exe /AddToPath=0 /RegisterPython=0 /S /D=C:\Users\<USER>\Miniconda\`

**Comment by ilanschnell** [comment]
Thanks for giving it a try, Steve!

**Comment by tkelman** [comment]
has miniconda changed anything that leads you to believe this is fixed in any other environment? one person being unable to reproduce is not usually a good reason to close an issue, unless you think you've made a change that fixed it. cc @davidanthoff and @brettcannon could you test again?

**Comment by msarahan** [comment]
@tkelman, we have updated our installers to use NSIS 3, rather than 2.46.  We had hoped that this would fix the issue.  We would welcome additional testing - @ilanschnell can you put Miniconda testing installers in a public place for @tkelman @davdanthoff @brettcannon?

**Comment by ilanschnell** [comment]
I just uploaded Miniconda installers (built using NSIS 3.0) here: 
http://repo.continuum.io/pkgs/misc/preview/

**Comment by ilanschnell** [comment]
I just uploaded Miniconda installers (built using NSIS 3.0) here: 
http://repo.continuum.io/pkgs/misc/preview/

**Comment by tkelman** [comment]
ah great, that sounds promising. didn't realize nsis had been upgraded, might've been worth saying that

**Comment by ilanschnell** [comment]
I think merging https://github.com/ContinuumIO/anaconda/issues/561 automatically closed this issue.  This was not intentional.  Reopening

**Comment by davidanthoff** [comment]
I just tried this with Miniconda3-4.2.5-Windows-x86_64.exe and it still crashes Outlook, doesn't seem to help.

**Comment by tkelman** [comment]
ah i see @ilanschnell, since that's a private repo others can't see the cross link.

**Comment by mattdmorgan** [comment]
Likewise, Miniconda3-4.2.5-Windows-x86_64.exe still crashes Outlook for me in the same way. Full flags used: '/InstallationType=JustMe /AddToPath=0 /RegisterPython=0 /NoRegistry=1 /S /D=C:\SomePath'

**Comment by msarahan** [comment]
Based on http://forums.winamp.com/archive/index.php/t-260624.html, I have attempted to change calls to ```Exec``` to using the ```ExecDos``` plugin with the ```/ToStack``` option, because of this comment from the email thread:

> I tried the exec2stack example for ExecDos, and it did *not* crash Outlook. I doubt it would sufer from the nsExec problem, since it does not access the listview window.

I am still seeing the crashes.  We will keep working on this, but our chances of resolving this for the next Anaconda release are dwindling.  The callstack for the crash ends at ```CLVItemStore::OnInsertItemA(tagLVITEMA*)``` - if that means anything to any of you, please make suggestions on where we should look.

**Comment by mingwandroid** [comment]
Hi all, I've fixed this bug with https://github.com/mingwandroid/nsis-untgz/commit/29e79565c3f434b6987c0067b1cc81be6bacacf8

We've verified the fix in our v4.2 test installers and the issue will not be present in the final release.

Github is going to close this issue on me, but I'll reopen it and close it again at final release.
